macrosynergy/dataquery/api.py

```diff
@@ -62,21 +62,18 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
             print(f'Execution {exc_type} with value (exc_value):\n{exc_value}')
 
     def check_connection(self) -> bool:
-        """Check connect (heartbeat) to DataQuery"""
-        js: dict = self.access.get_dq_api_result(url=self.access.base_url + "/services/heartbeat")
+        """Check connection (heartbeat) to DataQuery.
+        """
+        endpoint = "/services/heartbeat"
+        js: dict = self.access.get_dq_api_result(url=self.access.base_url + endpoint)
 
         results: dict = js["info"]
-        if int(results["code"]) != 200:
-            print(
-                f"DataQuery response {results['message']:s}"
-                f" with description: {results['description']:s}"
-            )
 
         return int(results["code"]) == 200
```

Comment:
If we are not printing error messages, we should at least return it:

```python
if int(results["code"]) != 200:
    msg = f"DataQuery response {results['message']:s} with description: {results['description']:s}"
else:
    msg = None

return int(results["code"]) == 200, msg
```

Reply:
Following an earlier comment, error message is printed where test_connection() is called.

Reply:
Will do.